include/camp/defines.hpp

```diff
@@ -21,7 +21,7 @@ For details about use and distribution, please read LICENSE and NOTICE from
 namespace camp
 {
 
-#define CAMP_ALLOW_UNUSED_LOCAL(X) (void)X
+#define CAMP_ALLOW_UNUSED_LOCAL(X) [&X]{}()
```

Comment:
Is this the new way to silence those warnings? Any pros/cons over something like `static_cast<void>(X)`? @trws?

Reply:
I tried with static_cast however I was still getting warnings with nvcc 😞 
I just gave each version a spin on [godbolt](https://godbolt.org/z/9xod8N), using intel,gcc and clang the lambda is less performant with -O0 but -O1 or higher optimizes all solutions away the same. msvc however doesn't seem to want to optimize the lambda away quite as nicely.

Reply:
Perhaps this might be better ?
`#if defined(__NVCC__)`
`#CAMP_ALLOW_UNUSED_LOCAL(X) [&X]{}()`
`#else `
`#CAMP_ALLOW_UNUSED_LOCAL(X) (void)X`
`#endif `

Reply:
Yea, that could be good if you were noticing some optimization differences.

Reply:
I'd rather avoid the lambda.  If we need something like this, how about using camp::sink, which I just realized I forgot to make the arguments const references, but either way that should pretty much always optimize out, and we can add some extra flags to it to make sure.  It's equivalent to what boost uses for their `ignore_unused` basically, like this:

```c++
template<typename...Ts>
<should be force inline> inline constexpr void sink(Ts const&...) {
}
```